images/kcp-registrar/register.sh

```diff
@@ -24,50 +24,50 @@ usage() {
 
     # Parameters
     printf "The following parameters need to be passed to the script:\n"
-    printf "KCP_ORG: the organistation for which the workload clusters need to be registered, i.e.: root:pipelines-service\n"
+    printf "KCP_ORG: the organization for which the workload clusters need to be registered, i.e.: root:pipelines-service\n"
     printf "KCP_WORKSPACE: the name of the workspace where the workload clusters get registered (created if it does not exist), i.e: compute\n"
     printf "DATA_DIR: the location of the cluster files\n"
     printf "          a single file with extension kubeconfig is expected in the subdirectory: credentials/kubeconfig/kcp\n"
     printf "          kubeconfig files for compute clusters are expected in the subdirectory: credentials/kubeconfig/compute \n\n"
 }
 
-prechecks () {
+exit_error() {
+    printf "[ERROR] %s" "$@" >&2
+    usage
+    exit 1
+}
+
+prechecks() {
     KCP_ORG=${KCP_ORG:-}
     if [[ -z "${KCP_ORG}" ]]; then
-        printf "KCP_ORG not set\n\n"
-        usage
-        exit 1
+        exit_error "KCP_ORG not set\n\n"
     fi
 
-    KCP_ORG=${KCP_WORKSPACE:-}
+    KCP_WORKSPACE=${KCP_WORKSPACE:-}
     if [[ -z "${KCP_WORKSPACE}" ]]; then
-        printf "KCP_WORKSPACE not set\n\n"
-        usage
-        exit 1
+        exit_error "KCP_WORKSPACE not set\n\n"
     fi
-    
+
     DATA_DIR=${DATA_DIR:-}
     if [[ -z "${DATA_DIR}" ]]; then
-        printf "DATA_DIR not set\n\n"
-        usage
-	exit 1
+        exit_error "DATA_DIR not set\n\n"
     fi
 }
 
 # populate kcp_kcfg with the location of the kubeconfig for connecting to kcp
 kcp_kubeconfig() {
-    if files=($(ls $DATA_DIR/credentials/kubeconfig/kcp/*.kubeconfig 2>/dev/null)); then
+    kubeconfig_dir="$DATA_DIR/credentials/kubeconfig"
+    if files=($(ls "$kubeconfig_dir/kcp/"*.kubeconfig 2>/dev/null)); then
         if [ ${#files[@]} -ne 1 ]; then
-            printf "A single kubeconfig file is expected at %s\n" "$DATA_DIR/credentials/kubeconfig/kcp"
-            usage
-            exit 1
+            exit_error "A single kubeconfig file is expected at $kubeconfig_dir/kcp\n\n"
         fi
-        kcp_kcfg="${files[0]}"
     else
-        printf "A single kubeconfig file is expected at %s\n" "$DATA_DIR/credentials/kubeconfig/kcp"
-        usage
-        exit 1
+        exit_error "A single kubeconfig file is expected at $kubeconfig_dir/kcp\n\n"
     fi
+    tmp_dir=$(mktemp -d)
+    cp -rf "$DATA_DIR/credentials" "$tmp_dir"
+    kubeconfig_dir="$tmp_dir/credentials/kubeconfig"
+    kcp_kcfg="$(ls "$kubeconfig_dir/kcp/"*.kubeconfig)"
```

Comment:
Nitpicking: 
You can as well copy the single kubeconfig file ${files[0]} in the if case. The above let to think that there are multiple kubeconfig files, which is not the case.
If you want to copy everything, it would be better to copy first and make the check for file uniqueness on the content of tmp_dir